RecoLocalMuon/GEMSegment/plugins/GEMSegmentBuilder.cc

```diff
@@ -27,6 +31,20 @@ GEMSegmentBuilder::GEMSegmentBuilder(const edm::ParameterSet& ps) : geom_(nullpt
 }
 GEMSegmentBuilder::~GEMSegmentBuilder() {}
 
+void GEMSegmentBuilder::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  edm::ParameterSetDescription ge0AlgoConfigDesc;
+  edm::ParameterSetDescription recAlgoConfigDesc;
+  desc.addOptional<bool>("enableGE0", true);
+  desc.addOptional<bool>("enableGE12", false);
```

Comment:
Thanks for the comments. I made it this way because I was running into issues when running workflows that the HLT couldn't run because it didn't have this parameter, and I found the optional elsewhere. I will try the fillDescriptions suggestion, I guess it should resolve if theres a default value?

Reply:
My understanding is that, if a plugin has a (static) `fillDescriptions` method with a parameter `add`ed with a default value, the plugin will use the default value of that parameter if said parameter is not explicitly included in the PSet of the python module (the given instance of the plugin in the configuration). Side note related to HLT: if the plugin has a `fillDescriptions` method, it's important that every parameter has a default value, because HLT normally uses the `cfi` produced via the `fillDescriptions` method to define which parameters are allowed for a given module (in the ConfDB GUI), and this works correctly on the HLT side only if a default value is specified (@cms-sw/hlt-l2 can correct me). Regarding `fillDescriptions` in general, here is a [link](https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideConfigurationValidationAndHelp) with documentation.

In short, I would try something like https://github.com/missirol/cmssw/commit/9e5b8ea713b21aa8a5d20c63de7db7afe53e7686 (the existing `cfi` file is removed, because generated automatically via `GEMSegmentProducer::fillDescriptions`, hopefully with the same exact content).